File: sample-java-jpa/src/main/java/hq/sombriks/sample/model/Person.java
package hq.sombriks.sample.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.PrePersist;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import lombok.Data;

@Data
@Entity
@Table(name = "person")
public class Person {

  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "person_id")
  private Integer id;

  @Column(name = "person_creation")
  @Temporal(TemporalType.TIMESTAMP)
  private Date creation;

  @Column(name = "person_name")
  private String name;

  @PrePersist
  public void preInsert() {
    if (creation == null)
      creation = new Date();
  }
}

File: sample-javascript-sequelize/src/model/Person.js
// model/Person.js
const Sequelize = require("sequelize");
const {sequelize} = require("../db");

exports.Person = sequelize.define(
  "Person",
  {
    id: {
      type: Sequelize.INTEGER,
      primaryKey: true,
      field: "person_id",
    },
    creation: {
      type: Sequelize.DATE,
      field: "person_creation",
    },
    name: {
      type: Sequelize.STRING,
      field: "person_name",
    },
  },
  {tableName: "person", timestamps: false},
);
